systemtest/src/test/java/io/strimzi/systemtest/KafkaClusterIT.java

```diff
@@ -382,4 +382,30 @@ public void testForTopicOperator() {
         List<String> topics = listTopicsUsingPodCLI(CLUSTER_NAME, kafkaPodName(CLUSTER_NAME, 1));
         assertThat(topics, not(hasItems("topic-from-cli", "my-topic")));
     }
+
+    @Test
+    @JUnitGroup(name = "regression")
+    @KafkaCluster(name = CLUSTER_NAME,
+            kafkaNodes = 1,
+            zkNodes = 1,
+            config = {
+                    @CmData(key = "kafka-rack",
+                            value = "{\"topologyKey\": \"rack-key\"}")
+            })
+    public void testRackAware() {
+
+        String cm = kubeClient.get("cm", CLUSTER_NAME);
+        assertThat(cm, valueOfCmEquals("kafka-rack", "{\"topologyKey\": \"rack-key\"}"));
+
+        kubeClient.waitForStatefulSet(kafkaClusterName(CLUSTER_NAME), 1);
+        String kafkaPodName = kafkaPodName(CLUSTER_NAME, 0);
+        kubeClient.waitForPod(kafkaPodName);
+
+        String out = kubeClient.exec(kafkaPodName, "cat /opt/kafka/rack/rack.id").out();
```

Comment:
Wouldn't it be better to also check the Kafka config file? That would check that the input (`/opt/kafka/rack/rack.id`) is properly used for the config.

Reply:
Good idea! I have done it as well.